src/timeout/timeout-collection.model.ts

```diff
@@ -1,16 +1,23 @@
 import { originalClearTimeout, originalSetTimeout } from '../overrides/override'
+import { TimeoutStatus } from './timeout-status.model';
 import { Timeout } from './timeout.model';
 
 export class TimeoutCollection {
 	private _timeoutCollection: Timeout[] = [];
 
 	public add(handler: any, timeout?: any, ...args: any[]) {
-		let id = originalSetTimeout.apply(window, [handler, timeout, args]);
-		this._timeoutCollection.push({id, handler, timeout, arguments: args, timestamp: Date.now()});
-
+		let id = originalSetTimeout.apply(window, [this._getWrappedHandler(handler), timeout, args]);
+		this._timeoutCollection.push(new Timeout(id, handler, timeout, args));
 		return id;
 	}
 
+	private _getWrappedHandler(handler: Function): Function {
+		return (() => {
+			this._timeoutCollection[this._getTimeoutIndexByHandler(handler)].status = TimeoutStatus.Completed;
+			return handler.apply(window);
```

Comment:
add a separating line before the return statement please

Reply:
I'm afraid I don't understand... If you give me a snippet of the desired code, I'll drop it in.

Reply:
Code

return blabla

I mean add an empty line between 16 and 17

Reply:
Okie dokie. I don't see that convention used elsewhere, but you're the boss!